src/query/src/dist_plan/analyzer.rs

```diff
@@ -100,8 +102,35 @@ impl AnalyzerRule for DistPlannerAnalyzer {
         plan: LogicalPlan,
         config: &ConfigOptions,
     ) -> datafusion_common::Result<LogicalPlan> {
-        // preprocess the input plan
-        let optimizer_context = OptimizerContext::new();
+        let mut config = config.clone();
+        // Aligned with the behavior in `datafusion_optimizer::OptimizerContext::new()`.
+        config.optimizer.filter_null_join_keys = true;
+        let config = Arc::new(config);
+
+        struct OptimizerContext {
```

Comment:
What is the purpose of this change?

Reply:
The `ConstEvaluator` in `SimplifyExpressions` might evaluate some UDFs early in the planning stage, by executing them directly. For example, the `database()` function. So the `ConfigOptions` here (which is set from the session context) should be present in the UDF's `ScalarFunctionArgs`. However, the default implementation in DataFusion seems to lost track on it: the `ConfigOptions` is recreated with its default values again. So we create a custom `OptimizerConfig` with the desired `ConfigOptions` to walk around the issue.

(also included in the code comments)

Reply:
Not sure if this is the best way. Should we create a DataFusion issue? WDYT @waynexia